src/main/java/frc/robot/RobotContainer.java

```diff
@@ -74,10 +74,16 @@ private void configureButtonBindings() {
         JoystickButton btnRotationSensitivity = new JoystickButton(rightJoystick, 1);
         JoystickButton btnIntakeBottomOut = new JoystickButton(altJoystick, 6);
 
-        driveSubsystem.setDefaultCommand(new RunCommand(() -> driveSubsystem.drive(altJoystick.getY(), altJoystick.getX()), driveSubsystem));
-        
-        shooterSubsystem.setDefaultCommand(new RunCommand(() -> shooterSubsystem.shooter(altJoystick.getThrottle()), shooterSubsystem));
-        btnIntakeIn.whileHeld(new RunCommand(() -> intakeSubsystem.spin(-1.0)));
+        driveSubsystem.setDefaultCommand(new RunCommand(() -> driveSubsystem.drive(-leftJoystick.getY() * 0.7, rightJoystick.getX() * 0.7), driveSubsystem));
+        launcherSolenoid
+            .whenPressed(new InstantCommand(() -> shooterSubsystem.activatePiston(), shooterSubsystem))
+            .whenInactive(new InstantCommand(() -> shooterSubsystem.lowerPiston(), shooterSubsystem));
+        shooterSubsystem.setDefaultCommand(new RunCommand(() -> {
+            shooterSubsystem.shooter(Math.abs(altJoystick.getY()));
+            System.out.println("Spinning");
+        }, shooterSubsystem));
+        btnIntakeOut.whenHeld(new InstantCommand(() -> intakeSubsystem.spin(0.5), intakeSubsystem)).whenInactive(new InstantCommand(() -> intakeSubsystem.spin(0), intakeSubsystem), true);
+        btnIntakeIn.whileHeld(new InstantCommand(() -> intakeSubsystem.spin(-0.5), intakeSubsystem)).whenInactive(new InstantCommand(() -> intakeSubsystem.spin(0), intakeSubsystem), true);
     }
```

Comment:
You should do the same thing as the launcherSolenoid and put the `whenX` on newlines.

Reply:
I feel like this "whenHeld" turn on "whenInactive" turn off pattern isn't going to scale well... I'm not sure if the solution is to just use variables inside the subsystems and reset them to 0 after each loop like in magicbot or to make the commands we use more state machine like.